spyder/plugins/editor/extensions/docstring.py

```diff
@@ -0,0 +1,517 @@
+            arg_text += '):'
+
+            if arg_value:
+                arg_value = arg_value.replace(self.quote3, self.quote3_other)
+                arg_text += ' Defaults to {}.'.format(arg_value)
+
+            arg_text += ' DESCRIPTION\n'
+
+        google_doc += arg_text
+
+        google_doc += '\n{}Returns:'.format(indent1)
+        if func_info.return_type:
+            google_doc += '\n{}{}: DESCRIPTION\n'.format(indent2,
+                                                         func_info.return_type)
+        else:
+            google_doc += '\n{}RETURN_TYPE: DESCRIPTION\n'.format(indent2)
+
+        google_doc += '\n{}{}'.format(indent1, self.quote3)
+
+        return google_doc
+
+
+class FunctionInfo:
+    """Parse function definition text."""
+
+    def __init__(self):
+        """."""
+        self.has_info = False
+        self.func_text = ''
+        self.args_text = ''
+        self.func_indent = ''
+        self.arg_name_list = []
+        self.arg_type_list = []
+        self.arg_value_list = []
+        self.return_type = None
+
+    @staticmethod
+    def is_char_in_pairs(pos_char, pairs):
+        """Return True if the charactor is in pairs of brackets or quotes."""
+        for pos_left, pos_right in pairs.items():
+            if pos_left < pos_char < pos_right:
+                return True
+
+        return False
+
+    @staticmethod
+    def find_quote_position(text):
+        """Return the start and end position of pairs of quotes."""
+        pos = {}
+        is_found_left_quote = False
+
+        for i, c in enumerate(text):
+            if is_found_left_quote is False:
+                if c == "'" or c == '"':
+                    is_found_left_quote = True
+                    quote = c
+                    left_pos = i
+            else:
+                if c == quote and text[i - 1] != '\\':
+                    pos[left_pos] = i
+                    is_found_left_quote = False
+
+        if is_found_left_quote:
+            raise IndexError("No matching close quote at: " + str(left_pos))
+
+        return pos
+
+    def find_bracket_position(self, text, bracket_left, bracket_right,
+                              pos_quote):
+        r"""Return the start and end position of pairs of brackets.
+
+        https://stackoverflow.com/questions/29991917/
+        indices-of-matching-parentheses-in-python
+        """
+        pos = {}
+        pstack = []
+
+        for i, c in enumerate(text):
+            if c == bracket_left and not self.is_char_in_pairs(i, pos_quote):
+                pstack.append(i)
+            elif c == bracket_right and not self.is_char_in_pairs(i,
+                                                                  pos_quote):
+                if len(pstack) == 0:
+                    raise IndexError(
+                        "No matching closing parens at: " + str(i))
+                pos[pstack.pop()] = i
+
+        if len(pstack) > 0:
+            raise IndexError(

```

Comment:
You catch these somewhere else rather than dumping them at the user, right?

Reply:
I changed find_bracket_position method to private method(_find_bracket_position).
This method is used only in split_args_text_to_list method.